tools/debug_tools/crash_log_parser/crash_log_parser.py

```diff
@@ -122,156 +104,117 @@ def parseUFSR(ufsr):
+        elif(eachline.startswith("HFSR")):
+            hfsr_val = parse_line_for_register(eachline)
             
-        if(eachline.startswith("MMFSR")):
-            l = re.findall(r"[\w']+", eachline)
-            l = [x.strip() for x in l if x != '']
-            tag, mmfsr_val = l
+        elif(eachline.startswith("MMFSR")):
+            mmfsr_val = parse_line_for_register(eachline)
             
-        if(eachline.startswith("BFSR")):
-            l = re.findall(r"[\w']+", eachline)
-            l = [x.strip() for x in l if x != '']
-            tag, bfsr_val = l
+        elif(eachline.startswith("BFSR")):
+            bfsr_val = parse_line_for_register(eachline)
             
-        if(eachline.startswith("UFSR")):
-            l = re.findall(r"[\w']+", eachline)
-            l = [x.strip() for x in l if x != '']
-            tag, ufsr_val = l
+        elif(eachline.startswith("UFSR")):
+            ufsr_val = parse_line_for_register(eachline)
             
-        if(eachline.startswith("CPUID")):
-            l = re.findall(r"[\w']+", eachline)
-            l = [x.strip() for x in l if x != '']
-            tag, cpuid_val = l
+        elif(eachline.startswith("CPUID")):
+            cpuid_val = parse_line_for_register(eachline)
             
-        if(eachline.startswith("MMFAR")):
-            l = re.findall(r"[\w']+", eachline)
-            l = [x.strip() for x in l if x != '']
-            tag, mmfar_val = l
+        elif(eachline.startswith("MMFAR")):
+            mmfar_val = parse_line_for_register(eachline)
             
-        if(eachline.startswith("BFAR")):
-            l = re.findall(r"[\w']+", eachline)
-            l = [x.strip() for x in l if x != '']
-            tag, bfar_val = l    
-        
+        elif(eachline.startswith("BFAR")):
+            bfar_val = parse_line_for_register(eachline)    
+    
+    print("\n\nCrash Info:")        
     print("\tCrash location = %s [%s] (based on PC value)"%(pc_name.strip(),str(pc_val)))
     print("\tCaller location = %s [%s] (based on LR value)"%(lr_name.strip(),str(lr_val)))        
     print("\tStack Pointer at the time of crash = [%s]"%(str(sp_val)))
     
-    print("\tTarget/Fault Info:")
-    parseCPUID(cpuid_val)
-    parseHFSR(hfsr_val)
-    parseMMFSR(mmfsr_val,mmfar_val)
-    parseBFSR(bfsr_val,bfar_val)
-    parseUFSR(ufsr_val)
-        
-    print("\nDone parsing...")    
+    print("\tTarget and Fault Info:")
+    print_CPUID_info(cpuid_val)
+    print_HFSR_info(hfsr_val)
+    print_MMFSR_info(mmfsr_val,mmfar_val)
+    print_BFSR_info(bfsr_val,bfar_val)
+    print_UFSR_info(ufsr_val)
         
                             
 if __name__ == '__main__':
     import argparse
     
-    parser = argparse.ArgumentParser(
-        description='Analyse mbed-os crash log')
-
+    parser = argparse.ArgumentParser(description='Analyse mbed-os crash log. This tool requires arm-gcc binary utilities to be available in current path as it uses \'nm\' command')
```

Comment:
Is this description good enough to warn the user that we need nm?

Reply:
I think so. You may want to call out that it's `arm-none-eabi-nm`.